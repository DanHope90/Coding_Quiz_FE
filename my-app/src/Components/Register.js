import axios from "axios";
import React, { useState } from "react";
import Alert from "./Alert";
import "../Styles/Register.css";

function Register() {
  const [formData, setFormData] = useState({
    userName: "",
    email: "",
    password: "",
    password2: "",
  });

  const alertState = {
    alert: {
      message: "",
      isSuccess: false,
    },
  };

  const [alert, setAlert] = useState(alertState.alert);

  const {
    userName, email, password, password2,
  } = formData;

  function addUser(event) {
    event.preventDefault();
    setAlert({ message: "", isSuccess: false });
    axios
      .post(`http://localhost:4000/api/user`, { ...formData })
      .then(() => {
        setAlert({
          message: "User has now been added!",
          isSuccess: true,
        });
      })
      .catch(() => {
        setAlert({
          message: "User already exists.",
          isSuccess: true,
        });
      });
  }

  function handleUserChange(event) {
    setFormData((prevState) => ({
      ...prevState,
      [event.target.name]: event.target.value,
    }));
  }

  return (
    <>
      <div>
        <h1 className="titles">Register</h1>
        <h2 className="titles">Please create an account</h2>
      </div>
      <div>
        <form className="form" onSubmit={addUser}>
          <Alert message={alert.message} success={alert.isSuccess} />
          <input
            type="text"
            className="form-control"
            id="userName"
            name="userName"
            value={userName}
            placeholder="Enter username"
            onChange={handleUserChange}
          />
          <input
            type="email"
            className="form-control"
            id="email"
            name="email"
            value={email}
            placeholder="Enter email address"
            onChange={handleUserChange}
          />
          <input
            type="password"
            className="form-control"
            id="password"
            name="password"
            value={password}
            placeholder="Enter password"
            onChange={handleUserChange}
          />
          <input
            type="password"
            className="form-control"
            id="password2"
            name="password2"
            value={password2}
            placeholder="Confirm password"
            onChange={handleUserChange}
          />
          <button type="submit" className="register-button">Submit</button>
        </form>
      </div>
    </>
  );
}

export default Register;
