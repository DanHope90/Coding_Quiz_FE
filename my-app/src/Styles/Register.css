.titles{
    display:flex;
    justify-content: center;
    flex-direction: column;
}

.form {
    display: flex;
    align-items: center;
    flex-direction: column;
    gap: 20px;
    padding-top: 20px;
}

.form-control {
    width: 400px;
}

.register-button {
    height: 20px;
    width: 200px;
    color: white;
}

.register-button.hover {
    color: red;
}

@media only screen and (max-width: 600px) {
    .form-control {
        width: 200px;
        font-size: 8px;
    }
}