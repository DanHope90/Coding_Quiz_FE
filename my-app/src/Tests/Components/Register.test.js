import React from "react";
import renderer from "react-test-renderer";
import Register from "../../Components/Register";

test("renders the Home component correctly", () => {
  const testRenderer = renderer.create(<Register />);
  expect(testRenderer).toMatchSnapshot();
});
