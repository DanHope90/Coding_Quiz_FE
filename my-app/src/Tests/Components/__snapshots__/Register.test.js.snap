// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Register component correctly 1`] = `
Array [
  <div>
    <h1>
      Register
    </h1>
    <h2>
      Please create an account
    </h2>
  </div>,
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="form-control"
        id="userName"
        name="userName"
        onChange={[Function]}
        placeholder="Enter username"
        type="text"
        value=""
      />
      <input
        className="form-control"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Enter email address"
        type="email"
        value=""
      />
      <input
        className="form-control"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
        value=""
      />
      <input
        className="form-control"
        id="password2"
        name="password2"
        onChange={[Function]}
        placeholder="Confirm password"
        type="password"
        value=""
      />
      <button
        className="button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
]
`;
